### PR TITLE
Migrate xpallets-gateway-common to pallet v2

### PR DESCRIPTION
**Pull request by Han-sx**

### CODE REVIEW

File: xpallets/gateway/common/src/lib.rs [hunk 1]
```diff
@@ -44,201 +42,103 @@ use self::types::{
...
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use frame_support::pallet_prelude::*;
+    use frame_system::pallet_prelude::*;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config + xpallet_gateway_records::Config {
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+        type Validator: Validator<Self::AccountId>;
```
Comment (0xrjman) [inline xpallets/gateway/common/src/lib.rs:57]:
```suggestion

        type Validator: Validator<Self::AccountId>;
```

File: xpallets/gateway/common/src/lib.rs [hunk 2]
```diff
@@ -295,14 +199,168 @@ decl_module! {
...
+        SetTrusteeProps(T::AccountId, Chain, GenericTrusteeIntentionProps),
+        /// An account set its referral_account of some chain. [who, chain, referral_account]
+        ReferralBinded(T::AccountId, Chain, T::AccountId),
+        /// The trustee set of a chain was changed. [chain, session_number, session_info]
+        TrusteeSetChanged(Chain, u32, GenericTrusteeSessionInfo<T::AccountId>),
+    }
+
+    /// Old name generated by `decl_event`.
+    #[deprecated(note = "use `Event` instead")]
+    pub type RawEvent<T> = Event<T>;
```
Comment (0xrjman) [inline xpallets/gateway/common/src/lib.rs]: There is no `RawEvent`, it's unnecessary.

File: xpallets/gateway/common/src/lib.rs [hunk 3]
```diff
@@ -337,7 +395,6 @@ impl<T: Config> Module<T> {
 }
 
 pub fn is_valid_about<T: Config>(about: &[u8]) -> DispatchResult {
-    // TODO
```
Comment (0xrjman) [inline xpallets/gateway/common/src/lib.rs]: There is no description in this `TODO` and it may not be completed, I am not sure whether it should be removed

**Review by 0xrjman — COMMENTED** [review]
The rest is LGTM

File: xpallets/gateway/common/src/lib.rs (hunk 3 above)
Comment (Han-sx) [reply xpallets/gateway/common/src/lib.rs]: Maybe you are right, the code exists in the old version of the file, I probably shouldn’t change it.

**Pull request merged**